analysis/code/calculate_road_distance_raster.R

```diff
@@ -0,0 +1,40 @@
+
+library(raster)
+library(fasterize)
+library(tidyverse)
+library(sf)
+library(fs)
+library(terra)
+
+# loading roads into memory
+road_files <- 'analysis/data/California Roads by County/' |>
+  fs::dir_ls(recurse = TRUE) |>
+  str_subset('.shp$')
+roads <- road_files |>
+  map(st_read) |>
+  bind_rows()
+
+# load an RRK metric for raster extent and resolution
+accel_fn <- 'analysis/data/Sierra Nevada ACCEL/fireAdaptComm/DamagePotential_WUI_2022_300m_normalized.tif'
+accel_extent <- raster(accel_fn)
+# set all values to zero (just to be safe)
+values(accel_extent) <- 0
+names(accel_extent) <- 'buildings_present'
+# set correct CRS
+proj4string(accel_extent) <- CRS("+init=EPSG:3310")
+
+# rasterize, checking if roads are present in each cell
+roads_raster <- roads %>%
+  st_transform(crs(accel_extent)) %>%
+  select() %>%
+  as_Spatial() %>%
+  rasterize(y = accel_extent)
+writeRaster(roads_raster, 'analysis/output/roads_raster.tif')
+
+# make into NA or 0 values
```

Comment:
what are the interpretations of NaN and 0 in this raster? For the RRK, we set NoData = NaN, whereas 0 is an actual value

Reply:
This raster should not contain any `NaN`s. I took the whole road layer and calculated distance from road polygons for ALL cells within the extent of the raster. If you tell me which mask to use I can mask it to RRK cells as well, but I did not want to mask cells based on the example raster layer I used because that seemed to be limited to locations with values for the specific metric, which may not be consistent across layers.

Reply:
For representation of a road, I think that depends on the front end representation or whatever ForSys needs. @thequeenofspades any insights on what would be better for the UI if we wanted to eventually display this?

No need to mask it to the RRKs! I just wanted to verify that if there were any NoDatas, that they were using our standard value.

Reply:
This will be represented as a raster/heatmap right? To me it would intuitively make sense for `1` to mean a cell contains a road rather than `0`, since `1` feels like the "hot" value.

Reply:
Thanks @thequeenofspades,
One clarification in case I didn't communicate this well: cells WITHOUT roads will have values > 0 (or > 1 if we change roads to 1) since this raster indicates distance from road rather than presence / absence of road.

Reply:
Oh I missed that. In that case `0` for cells with roads makes sense!